apps/api-extractor/src/generators/PackageTypingsGenerator.ts

```diff
@@ -0,0 +1,402 @@
+          if (parentSpan) {
+            switch (parentSpan.kind) {
+              case ts.SyntaxKind.ExpressionWithTypeArguments:
+              case ts.SyntaxKind.TypeReference:
+
+              case ts.SyntaxKind.ClassDeclaration:
+              case ts.SyntaxKind.InterfaceDeclaration:
+              case ts.SyntaxKind.EnumDeclaration:
+              case ts.SyntaxKind.TypeAliasDeclaration:
+              case ts.SyntaxKind.ModuleDeclaration:  // (namespaces are a type of module declaration)
+                {
+                  const symbol: ts.Symbol | undefined = this._typeChecker.getSymbolAtLocation(span.node);
+                  if (!symbol) {
+                    throw new Error('Symbol not found');
+                  }
+
+                  const referencedEntry: Entry | undefined = this._fetchEntryForSymbol(symbol);
+                  if (referencedEntry) {
+                    if (!referencedEntry.uniqueName) {
+                      // This should never happen
+                      throw new Error('referencedEntry.uniqueName is undefined');
+                    }
+
+                    span.modification.prefix = referencedEntry.uniqueName;
+                    // span.modification.prefix += '/**/';
+                  }
+                }
+                break;
+            }
+          }
+          break;
+        }
+      }
+    );
+  }
+
+  /**
+   * Ensures a unique name for each item in the package typings file.
+   */
+  private _makeUniqueNames(): void {
+    const usedNames: Set<string> = new Set<string>();
+    for (const entry of this._entries) {
+      let suffix: number = 1;
+      entry.uniqueName = entry.localName;
+      while (usedNames.has(entry.uniqueName)) {
+        entry.uniqueName = entry.localName + '_' + ++suffix;
+      }
+      usedNames.add(entry.uniqueName);
+    }
+  }
+
+  private _fetchEntryForSymbol(symbol: ts.Symbol): Entry | undefined {
+    const result: IFollowAliasesResult = PackageTypingsGenerator._followAliases(symbol, this._typeChecker);
+    if (result.external || !result.moduleExport) {
+      return; // external definition
+    }
+
+    const followedSymbol: ts.Symbol = result.symbol;
+    if (followedSymbol.flags & (
+      ts.SymbolFlags.TypeParameter | ts.SymbolFlags.TypeLiteral
+      )) {
+      return undefined;
+    }
+
+    let entry: Entry | undefined = this._entriesBySymbol.get(followedSymbol);
+    if (entry) {
+      return entry;
+    }
+
+    entry = new Entry();
+    entry.localName = symbol.name;
+    entry.followedSymbol = followedSymbol;
+    this._entries.push(entry);
+    this._entriesBySymbol.set(followedSymbol, entry);
+    console.log('======> ' + entry.localName);
+
+    for (const declaration of followedSymbol.declarations || []) {
+      // console.log(PrettyPrinter.dumpTree(declaration));
+      // console.log('-------------------------------------');
+      // console.log(declaration.getText());
+      // console.log('=====================================');
```

Comment:
Clean up all of this commented out code.

Reply:
concur.

---
In reply to: [152428250](https://github.com/Microsoft/web-build-tools/pull/448#discussion_r152428250) [](ancestors = 152428250)